src/Common/dot_wrapper.h

```diff
@@ -0,0 +1,29 @@
+#ifndef DOT_WRAPPER_H
+#define DOT_WRAPPER_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#ifdef _WIN32
+#define inline __inline
+#endif
+
+#ifdef CARET_DOTFCN
+#include "dot.h"
```

Comment:
One additional thing - we may want a runtime setting to switch to the naive implementation, rather than recompiling (for easier debugging, especially for end users).  Maybe something like a "dot_set_impl(int some_enum)" function that can set naive, or sse2 despite avx support, and reset to the usual selection (so that in the GUI it can be toggled off and then back on).  The enum could have a "DOT_BEST" value for future-proofing (could just reinitialize it to the select function, and not repeat the logic), as well as "DOT_NAIVE", "DOT_SSE2", "DOT_AVX".  It could return a bool or condition code indicating whether the requested mode was supported by the cpu features (and for simplicity of state tracking, set to naive if it wasn't supported?).  If you wanted to avoid duplication of the cpuid checks, you could even rewrite the select functions in terms of calls to this new function.

We can deal with adding the command line switches and preference settings that make use of such a function.